spyder/widgets/panels/edgeline.py

```diff
@@ -49,11 +48,23 @@ def set_enabled(self, state):
         self._enabled = state
         self.setVisible(state)
 
-    def set_column(self, column, index=0):
-        """Set edge line column value"""
-        self.columns[index] = column
+    def set_columns(self, columns):
+        """Set edge line columns values."""
+        if isinstance(columns, list):
+            self.columns = columns
+        elif isinstance(columns, int):
+            self.columns = [columns]
+        elif isinstance(columns, str):
```

Comment:
hmm no, It was necessary when there wasn't the regex validation, I think this validation could be deleted and use just `else`